extension/extensionlimiter/rate.go

```diff
@@ -0,0 +1,72 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package extensionlimiter // import "go.opentelemetry.io/collector/extension/extensionlimiter"
+
+import (
+	"context"
+)
+
+// RateLimiterProvider is a provider for rate limiters.
+//
+// Limiter implementations will implement this or the
+// ResourceLimiterProvider interface, but MUST not implement both.
+// Limiters are covered by configmiddleware configuration, which is
+// able to construct LimiterWrappers from these providers.
+type RateLimiterProvider interface {
+	CheckerProvider
+
+	// GetRateLimiter returns a rate limiter for a weight key.
+	GetRateLimiter(WeightKey, ...Option) (RateLimiter, error)
```

Comment:
Do you think we should have different `Option`s for different providers (different types so we can expand them independently)?

Reply:
I was thinking about passing in Signal kind, Component ID via these options. I wouldn't mind making these two parameters obligatory, but I don't know how to make it so--the confighttp and configgrpc wrappers don't have this info either.